### PR TITLE
strconv: add configurable thousands formatting

### PR DESCRIPTION
**Pull request by ChrizzzDev**
## Summary

Add thousands-separator formatting utilities to `strconv`.

This PR addresses issue #28765.

## What this PR does

- Add `strconv.format_thousands` for formatting built-in integer and floating-point values.
- Add `strconv.add_thousands_sep` for formatting numeric strings that are already converted to text.
- Support simple string separators for backwards-compatible usage.
- Add `strconv.Separator` to configure integer and decimal separators independently.
- Validate supported numeric types at compile time.
- Document the new API in the `strconv` README.
- Add focused tests covering integers, floats, signs, decimal separators, custom separators, and edge cases.

## Testing

- `v -silent test vlib/strconv/format_thousands_test.v`
- `v test-all`

### CODE REVIEW

**Review by medvednikov — COMMENTED** [review]
Reviewed all three changed files at `176f2bbe5fcf290cb54a6d6f5790a32eb3d1ba12`, excluding CI status. The latest patch preserves explicit exponents in the string API and adds both float-type assertions. Two remaining correctness findings:

### [P2] Preserve fixed-point formatting in the JS numeric path

In [`format_thousands`, lines 141–142](https://github.com/vlang/v/blob/176f2bbe5fcf290cb54a6d6f5790a32eb3d1ba12/vlib/strconv/format_thousands.v#L141-L142), the JS fallback uses `number.str()`. The existing [JS float implementation](https://github.com/vlang/v/blob/176f2bbe5fcf290cb54a6d6f5790a32eb3d1ba12/vlib/builtin/js/float.js.v) performs `x.val + ''`, so `f64(1e21)` becomes `1e+21`. Since `add_thousands_sep` now correctly preserves the exponent, `format_thousands(f64(1e21), ',')` consequently returns `1e+21` instead of grouping the full decimal integer part as the C float path does. This makes the numeric API stop adding thousands separators for large JS floats. Please provide equivalent fixed-point conversion/expand exponent notation in the JS **numeric** path, while retaining exponent preservation in the string API, and add a JS regression around `1e21`.

### [P2] Pass non-finite float representations through without grouping

`add_thousands_sep` passes the entire mantissa to `insert_thousands_sep` without checking whether it consists of digits. This matters for supported float inputs, not just arbitrary invalid strings: the same JS float conversion produces `Infinity` and `-Infinity`, which this algorithm turns into `In,fin,ity` and `-In,fin,ity` with a comma separator. The behavior can also be exercised directly with `add_thousands_sep('Infinity', ',')`. Please preserve recognized non-finite values before digit grouping and cover positive/negative infinity through the JS numeric API.

Validation: inspected the PR and surrounding C/JS conversion implementations, and reproduced the above JS-path results in Node using the existing native number-to-string operation and an equivalent implementation of the PR's grouping algorithm. This was **not** a compiled V test run: no V compiler was available locally, and the attempted repository checkout was blocked by network access.

**Review by medvednikov — COMMENTED** [review]
Follow-up review at `959dd33282d7e3916ca04fb5b9e9dd5fb3c15350`, excluding CI status.

Both original implementation findings are addressed in this revision: the JS numeric float path expands exponent notation before grouping (while the string API still preserves explicit exponents), and the digit guard prevents grouping `Infinity` / `-Infinity` into chunks.

One remaining P2 test-portability issue is noted inline: the new non-finite numeric assertions hard-code C-backend spellings and therefore disagree with the corrected JS output. Please make those expectations backend-aware or assert preservation of the value's native `.str()` representation.

Validation: inspected all three changed files and the JS float conversion at this head. A direct JavaScript translation of the updated helpers passed 29 explicit checks and 5,057 finite-value round-trip checks in Node, including large/small exponents and non-finite pass-through. It also reproduced the three non-finite assertion mismatches described inline. These were algorithm checks, **not compiled V tests**: no local V compiler was available, and repository access from the execution environment failed DNS resolution.

File: vlib/strconv/format_thousands_test.v [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	// `.str()` (on either the C or the JS backend) emits scientific
+	// notation; format_thousands() must still group the full integer part.
+	assert format_thousands(f64(1e21), ',') == '1,000,000,000,000,000,000,000.0'
+	assert format_thousands(1234567891.42, '.') == '1.234.567.891.42'
+}
+
+fn test_format_thousands_float_non_finite() {
+	assert format_thousands(math.inf(1), ',') == '+inf'
+	assert format_thousands(math.inf(-1), ',') == '-inf'
+	assert format_thousands(math.nan(), ',') == 'nan'
```
Comment (medvednikov) [inline vlib/strconv/format_thousands_test.v]: [P2] Make the non-finite regression expectations backend-aware

These assertions require `+inf`, `-inf`, and `nan` on every backend, but the JS float implementation (`vlib/builtin/js/float.js.v`) uses `x.val + ''`, producing `Infinity`, `-Infinity`, and `NaN`. The updated `expand_exponent` / `add_thousands_sep` path now correctly preserves those strings, so all three expectations mismatch when exercising the JS backend—the backend this fix targets.

Keep the pass-through behavior and make the test portable, for example by comparing each formatted non-finite value with its native representation:

```v
for value in [math.inf(1), math.inf(-1), math.nan()] {
    assert format_thousands(value, ',') == value.str()
}
```

Alternatively, use separate `$if js` expectations. This is a test-expectation issue, not a request to restore or normalize away the corrected JS pass-through behavior.

File: vlib/strconv/format_thousands_test.v [hunk 2]
```diff
@@ -60,7 +60,13 @@ fn test_format_thousands_float_large_magnitude() {
...
+	// Backend-agnostic on purpose: assert the pass-through property (a
+	// non-finite value's own `.str()` is never touched by grouping) rather
+	// than hard-coding one backend's spelling. On the C backend `.str()`
+	// gives '+inf'/'-inf'/'nan'; on the JS backend it gives
+	// 'Infinity'/'-Infinity'/'NaN'. Either way, format_thousands() must
+	// leave it exactly as `value.str()` produced it.
+	for value in [math.inf(1), math.inf(-1), math.nan()] {
+		assert format_thousands(value, ',') == value.str()
+	}
+}
```
Comment (JalonSolov) [inline vlib/strconv/format_thousands_test.v:72]: Missing EOL on last line of file (that's what the red circle with a `-` in it means).
